network-api/networkapi/buyersguide/tests.py

```diff
@@ -440,18 +460,455 @@ def name_change_changes_slug(self):
+        self.check_software_products_match()
+
+
+class TestProductPage(BuyersGuideTestMixin):
+
+    def setUp(self):
+        super().setUp()
+        if not hasattr(self.product_page.votes, 'get_votes'):
+            votes = ProductPageVotes()
+            votes.save()
+            self.product_page.votes = votes
+            self.product_page.save()
+
+    def test_get_votes(self):
+        # Votes should be empty at this point.
+        votes = self.product_page.votes.get_votes()
+        self.assertEqual(votes, [0, 0, 0, 0, 0])
+        self.assertEqual(len(votes), 5)
+
+    def test_set_votes(self):
+        # Make sure votes are set and saved.
+        self.product_page.votes.set_votes([1, 2, 3, 4, 5])
+        votes = self.product_page.votes.get_votes()
+        self.assertEqual(votes, [1, 2, 3, 4, 5])
+
+        # Ensure there's always 5 value set.
+        self.product_page.votes.set_votes([1, 2, 3, 4, 5, 6, 7])
+        self.assertEqual(votes, [1, 2, 3, 4, 5])
+        self.assertEqual(len(votes), 5)
+
+    def test_current_tally(self):
+        self.product_page.votes.set_votes([5, 4, 3, 2, 1])
+        current_tally = self.product_page.current_tally
+        self.assertEqual(current_tally, 15)
+
+        self.product_page.votes.set_votes([5, 5, 5, 5, 5])
+        current_tally = self.product_page.current_tally
+        self.assertEqual(current_tally, 25)
+
+    def test_creepiness(self):
+        self.product_page.current_vote_count = 100
+        self.product_page.votes.set_votes([5, 5, 5, 5, 5])
+        creepiness = self.product_page.creepiness
+        self.assertEqual(creepiness, 4)
+
+        self.product_page.current_vote_count = 0
+        self.product_page.votes.set_votes([0, 0, 0, 0, 0])
+        creepiness = self.product_page.creepiness
+        self.assertEqual(creepiness, 0)
+
+    def test_get_voting_json(self):
+        self.product_page.current_vote_count = 60
+        self.product_page.votes.set_votes([1, 2, 3, 4, 5])
+        creepiness = self.product_page.creepiness
+        self.assertEqual(creepiness, 4)
+
+        current_tally = self.product_page.current_tally
+        self.assertEqual(current_tally, 15)
+
+        # votes = self.product_page.votes.get_votes()
+        data = json.loads(self.product_page.get_voting_json)
+        comparable_data = {
+            'creepiness': {
+                'vote_breakdown':  {
+                    '0': 1,
+                    '1': 2,
+                    '2': 3,
+                    '3': 4,
+                    '4': 5,
+                },
+                'average': 4.0,
+            },
+            'total': 15,
+        }
+        self.assertDictEqual(data, comparable_data)
+
+    def test_get_or_create_votes(self):
+        # Delete potential votes
+        self.product_page.votes.delete()
+        self.product_page.votes = None
+        self.product_page.save()
+        self.assertFalse(hasattr(self.product_page.votes, 'set_votes'))
```

Comment:
Let's test for what we actually want to know, explicitly checking whether it's `None`, or whether it has the correct class. No need for duck typing in tests.